### ISSUE TITLE
Feature Request: Default platform setting

### ISSUE THREAD

**Issue by Malaber** [ben-gibson/GitLink#276]
Hey, first of all I love using GitLink!

It would be nice to set a default Platform as I have to set it to GitLab on every new project I open the first time (which is quite a lot sadly) 
Even cooler would be an option to set the default per folder, as I keep my locally cloned repositories separated by git server.

Thanks for considering this.

**Comment by dwightmulcahy** [comment]
Same, we recently moved to GitLab and having to change that for each repo is annoying...

**Comment by ben-gibson** [comment]
In the plugin settings (`Settings -> Tools -> GitLink -> Domain Registry`) you can register custom domains against each host. If you add the domain your using for GitLab, it will automatically detect `GitLab` as the host when opening a new project.

**Comment by Malaber** [comment]
Oh, yeah, I found that afterwards but forgot to close this thread. Thanks for the help, maybe others will find it better now. 

Also: it does not seem to sync to other IDEs via the official settings sync. (I had to register our gitlab in every IDE manually) This is really no big deal, but maybe that can be changed?

**Comment by dwightmulcahy** [comment]
Yeah, I noticed I was using 3.3.6 from the IntelliJ Marketplace instead of manually installing the latest version... 

Move along; nothing to see here.

**Comment by gelin** [comment]
It's nice to have a domain alias. For example, in case of GitHub I have to use an alias in `~/.ssh/config`:

```
Host github-work
   HostName github.com
   IdentityFile ~/.ssh/work
   IdentitiesOnly yes
```

In this case GitLink gives URLs like `https://github-work/...`. However, it must be `https://github.com/...`.

GitHub always sits in `github.com`. I just want GitLink to understand that `github-work` in my case is an alias for `github.com`.
